### DISCUSSION TITLE
Supermuc Seissol MPI Error

### DISCUSSION THREAD

**Discussion by jwjeremy** [SeisSol/SeisSol#609] [Q&A] [answered]
I am running the seissol in supermuc and encountered the following MPI error after submitting the job via sbatch. 

`Abort(671695877) on node 10 (rank 10 in comm 0): Fatal error in PMPI_Comm_rank: Invalid communicator, error stack:
PMPI_Comm_rank(107): MPI_Comm_rank(comm=0xf8cab300, rank=0x7f52f8fcce00) failed
PMPI_Comm_rank(64).: Invalid communicator
Abort(537478149) on node 1 (rank 1 in comm 0): Fatal error in PMPI_Comm_rank: Invalid communicator, error stack:
PMPI_Comm_rank(107): MPI_Comm_rank(comm=0x91b91300, rank=0x7fa991eb2e00) failed
PMPI_Comm_rank(64).: Invalid communicator
Abort(403260421) on node 5 (rank 5 in comm 0): Fatal error in PMPI_Comm_rank: Invalid communicator, error stack:
PMPI_Comm_rank(107): MPI_Comm_rank(comm=0xc6c81300, rank=0x7f7cc6fa2e00) failed
PMPI_Comm_rank(80).: Null Comm pointer
Abort(403260421) on node 3 (rank 3 in comm 0): Fatal error in PMPI_Comm_rank: Invalid communicator, error stack:
PMPI_Comm_rank(107): MPI_Comm_rank(comm=0x903bf300, rank=0x7f09906e0e00) failed
PMPI_Comm_rank(64).: Invalid communicator`

I think it is about the differences between the MPI version used in compilation and execution. However, the mpi version during compilation seems the same as the mpiexec.

Thank you so much!

**Comment by sebwolf-de** [comment]
Please use our issue template: 

**Describe the bug**
A clear and concise description of what the bug is. Tell us what the code actually does.

**Expected behavior**
A clear and concise description of what you expected to happen. Tell us what you want the code to do.

**To Reproduce**
Steps to reproduce the behavior:
1. Which version do you use? Provide branch and commit id.
2. Which build settings do you use? Which compiler version do you use? 
3. On which machine does your problem occur? If on a cluster: Which modules are loaded?
4. Provide parameter/material files.

**Screenshots/Console output**
If you suspect a problem in the numerics/physics add a screenshot of your output.

If you encounter any errors/warnings/... during execution please provide the console output.

**Additional context**
Add any other context about the problem here.

**Reply by jwjeremy** [reply]
Sorry for the inconvenience made!

I got an error during running seissol in supermuc cluster. The error is in the output of the job submission of the sbatch stating that there is some problem with the MPI issue when executing the seissol and the error message is attached.

`[Abort(671695877) on node 10 (rank 10 in comm 0): Fatal error in PMPI_Comm_rank: Invalid communicator, error stack: PMPI_Comm_rank(107): MPI_Comm_rank(comm=0xf8cab300, rank=0x7f52f8fcce00) failed PMPI_Comm_rank(64).: Invalid communicator Abort(537478149) on node 1 (rank 1 in comm 0): Fatal error in PMPI_Comm_rank: Invalid communicator, error stack: PMPI_Comm_rank(107): MPI_Comm_rank(comm=0x91b91300, rank=0x7fa991eb2e00) failed PMPI_Comm_rank(64).: Invalid communicator Abort(403260421) on node 5 (rank 5 in comm 0): Fatal error in PMPI_Comm_rank: Invalid communicator, error stack: PMPI_Comm_rank(107): MPI_Comm_rank(comm=0xc6c81300, rank=0x7f7cc6fa2e00) failed PMPI_Comm_rank(80).: Null Comm pointer Abort(403260421) on node 3 (rank 3 in comm 0): Fatal error in PMPI_Comm_rank: Invalid communicator, error stack: PMPI_Comm_rank(107): MPI_Comm_rank(comm=0x903bf300, rank=0x7f09906e0e00) failed PMPI_Comm_rank(64).: Invalid communicator]`


The seissol is successfully compiled a few days ago with the last commit of 4357f5e0 and it shows successfully reading the parameter files when tested in the terminal.  The error message shows there is some issue with the mpi in seissol. Some online resources state that the error is based on a different mpi version between the mpi used in compilation and the mpiexec. The mpiexec used in my account is 22.2.1 and the mpif90, mpicc versions are both 2021.4.0. I wonder if this cause the error.

The following attached are the list of modules used, the sbatch file used in running the seissol and bashrc used supermuc. 
[module_list.txt](https://github.com/SeisSol/SeisSol/files/9090962/module_list.txt)

[bashrc.txt](https://github.com/SeisSol/SeisSol/files/9090875/bashrc.txt)

[sbatch.slum.txt](https://github.com/SeisSol/SeisSol/files/9090871/sbatch.slum.txt)

Many thanks!

**Reply by sebwolf-de** [reply]
Can you output the result of `which mpiexec` with your current setup and with the python module unloaded. It seems to me that the python module provides their own mpiexec. Also in your first two lines of the bashrc you first set `SLURM_EAR_LOAD_MPI_VERSION` to `intel` and then you set it to `open mpi`, so effectively you have chosen `open mpi`.
I recommend to use `salloc -t 0:30:00 -p test -A <your compute project> -N 4 --ear=off` to get an interactive shell with 4 nodes. Then you can try out all different modules and `mpirun`, `mpiexec`, `srun` to find the solution, which works for you.

**Reply by krenzland** [reply]
See also the following note:
https://doku.lrz.de/display/PUBLIC/Energy+Aware+Runtime
regarding the SLURM_EAR_LOAD_MPI_VERSION

"(warning) Note: when combining a python-mpi and a regular mpi application (i.e., no python but C/C++/Fortran) in the same batch please unset this variable for the regular mpi application while using EAR, otherwise your application may crash:"

You can also try to deactivate EAR by adding
"#SBATCH --ear=off"
into your slurm batch script. This leads to a loss of performance, but EAR is super unstable.

**Reply by Thomas-Ulrich** [reply]
note also that you use potentially conflicting modules:
libszip/2.1.1, netcdf-hdf5-all/4.7_hdf5-1.10-intel21-impi, yaml-cpp/0.7.0-intel21, parmetis and metis are all duplicated in seissol-env.
I would unload all these modules.

**Reply by jwjeremy** [reply]
The path of the `mpiexec` is `/dss/lrzsys/sys/spack/release/22.2.1/opt/x86_64/intel-mpi/2019.12.320-gcc-wx7cjlg/compilers_and_libraries_2020.4.320/linux/mpi/intel64/bin/mpiexec`, hence i updated the `SLURM_EAR_LOAD_MPI_VERSION=intel` in the bashrc. The python module is also unloaded. However, `mpirun`, `mpiexec` and `srun` resulted the same MPI error. 

The `#SBATCH --ear=off` option successfully resolves the MPI error! And all the possible conflicting modules are unloaded.

Many thanks!

**Comment by sebwolf-de** [comment, marked as answer]
> The `#SBATCH --ear=off` option successfully resolves the MPI error! And all the possible conflicting modules are unloaded.

**Reply by Thomas-Ulrich** [reply]
@jwjeremy maybe you could open an LRZ ticket to have it fixed.
